portal/src/graphql/portal/DesignScreen/form.ts

```diff
@@ -540,6 +545,11 @@ export function useBrandDesignForm(appID: string): BranchDesignForm {
         ),
         image
       );
+      resourceMutator.updateCustomisableTheme((prev) => {
+        return produce(prev, (draft) => {
+          draft.logo.alignment = image == null ? "hidden" : undefined;
+        });
+      }, targetTheme);
```

Comment:
The problem here is because resourceForm.setState is passing the captured state instead of the latest state.

Please try the following fix.

1. https://github.com/authgear/authgear-server/blob/ba91f02fa57f6986cd79f2164fc9d0bb2159bff3/portal/src/hook/useResourceForm.ts#L152
```diff
  const state = currentState ?? initialState;
  const setState = useCallback(
    (fn: (state: State) => State) => {
+      setCurrentState(s => fn(s ?? initialState));
-      setCurrentState(fn(state));
    },
+    [initialState]
-    [state]
  );
```

Reply:
I think `useAppConfigForm` and `useAppSecretConfigForm` suffer the same problem. If the fix I purposed works, please help also update them